third_party/llvm-project/DWARFEmitter.cpp

```diff
@@ -184,6 +187,16 @@ class DumpVisitor : public DWARFYAML::ConstVisitor {
     }
   }
 
+  // XXX BINARYEN Make sure we emit the right size. We should not change the
+  // size as we only modify relocatable fields like addresses, and such fields
+  // have a fixed size, so any change is a bug.
+  void onEndCompileUnit(const DWARFYAML::Unit &CU) {
+    size_t EndPos = OS.tell();
+    if (EndPos - StartPos != CU.Length.getLength()) {
+      llvm_unreachable("compile unit size was incorrect");
```

Comment:
Is this intended to be debugging-only (i.e. only enabled in asserts builds)?

Reply:
I'd prefer it in release builds too, this is intentional - this is a very serious internal error that we should not skip.

Reply:
Right, I guess to clarify, IIRC `llvm_unreachable` will not work in release builds (it's a fancy `__builtin_unreachable`). You want something like LLVM's `report_fatal_error`.

Reply:
Oh, I see, thanks. I'll fix in a followup.

Reply:
Actually since we default to assertions builds, and test them, that's good enough coverage, and we can leave no-assertion builds to not assert on this.